frontend/src/Search.elm

```diff
@@ -1039,33 +1041,49 @@ viewResults nixosChannels model result viewSuccess _ outMsg categoryName =
 
         total =
             String.fromInt result.hits.total.value
+
     in
     [ div []
-        [ Html.map outMsg <| viewSortSelection model
-        , h2 []
-            (List.append
-                [ text "Showing results "
-                , text from
-                , text "-"
-                , text to
-                , text " of "
-                ]
-                (if result.hits.total.value == 10000 then
-                    [ text "more than 10000."
-                    , p [] [ text "Please provide more precise search terms." ]
-                    ]
-
-                 else
-                    [ strong []
-                        [ text total
-                        , text " "
-                        , text categoryName
-                        ]
+        (List.append
+          [ Html.map outMsg <| viewSortSelection model
+          , h2 []
+              (List.append
+                  [ text "Showing results "
+                  , text from
+                  , text "-"
+                  , text to
+                  , text " of "
+                  ]
+                  (if result.hits.total.value == 10000 then
+                      [ text "more than 10000."
+                      , p [] [ text "Please provide more precise search terms." ]
+                      ]
+
+                   else
+                      [ strong []
+                          [ text total
+                          , text " "
+                          , text categoryName
+                          ]
+                      , text "."
+                      ]
+                  )
+              )
+          ]
+          (case List.head result.hits.hits of
+            Nothing -> []
+            Just elem ->
+              case Array.get 3 (Array.fromList (String.split "-" elem.index)) of
+                Nothing -> []
+                Just commit ->
+                  [
+                    text "Data from nixpkgs "
+                    , a [ href ("https://github.com/NixOS/nixpkgs/commit/" ++ commit) ]
+                    [ (code [] [ text (String.slice 0 12 commit) ]) ]
```

Comment:
```suggestion
                    [ (code [] [ text (String.slice 0 8 commit) ]) ]
```

I suggest we use a length of 8 since that is what is the length of an abbreviated commit (see https://git-scm.com/docs/git-log#Documentation/git-log.txt---abbrev-commit).